File: src/main/java/com/dh/ondot/schedule/domain/enums/SnoozeInterval.java
package com.dh.ondot.schedule.domain.enums;

import com.dh.ondot.core.exception.UnsupportedException;
import lombok.Getter;

import java.util.Locale;

import static com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_SNOOZE_INTERVAL;

@Getter
public enum SnoozeInterval {
    ONE(1),
    THREE(3),
    FIVE(5),
    TEN(10),
    THIRTY(30),
    SIXTY(60);

    private final int value;

    SnoozeInterval(int value) {
        this.value = value;
    }

    public static SnoozeInterval from(String interval) {
        try {
            return SnoozeInterval.valueOf(interval.toUpperCase(Locale.ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException(UNSUPPORTED_SNOOZE_INTERVAL, interval);
        }
    }

    public static SnoozeInterval from(int value) {
        for (SnoozeInterval interval : values()) {
            if (interval.value == value) {
                return interval;
            }
        }
        throw new UnsupportedException(UNSUPPORTED_SNOOZE_INTERVAL, String.valueOf(value));
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/enums/SnoozeInterval.kt
package com.dh.ondot.schedule.domain.enums

import com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_SNOOZE_INTERVAL
import com.dh.ondot.core.exception.UnsupportedException
import java.util.Locale

enum class SnoozeInterval(val value: Int) {
    ONE(1),
    THREE(3),
    FIVE(5),
    TEN(10),
    THIRTY(30),
    SIXTY(60);

    companion object {
        @JvmStatic
        fun from(interval: String): SnoozeInterval =
            try {
                valueOf(interval.uppercase(Locale.ENGLISH))
            } catch (e: IllegalArgumentException) {
                throw UnsupportedException(UNSUPPORTED_SNOOZE_INTERVAL, interval)
            }

        @JvmStatic
        fun from(value: Int): SnoozeInterval =
            entries.firstOrNull { it.value == value }
                ?: throw UnsupportedException(UNSUPPORTED_SNOOZE_INTERVAL, value.toString())
    }
}
